symphony-bdk-core/src/main/java/com/symphony/bdk/core/service/datafeed/impl/DatafeedServiceV1.java

```diff
@@ -0,0 +1,138 @@
+package com.symphony.bdk.core.service.datafeed.impl;
+
+import com.symphony.bdk.core.api.invoker.ApiClient;
+import com.symphony.bdk.core.api.invoker.ApiException;
+import com.symphony.bdk.core.auth.AuthSession;
+import com.symphony.bdk.core.config.model.BdkConfig;
+import com.symphony.bdk.gen.api.model.V4Event;
+import lombok.extern.slf4j.Slf4j;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+@Slf4j
+public class DatafeedServiceV1 extends AbstractDatafeedService {
+
+    private static final String DATAFEED_ID_FILE = "datafeed.id";
+    private final AtomicBoolean started = new AtomicBoolean();
+    private String datafeedId;
+
+
+    public DatafeedServiceV1(ApiClient agentClient, ApiClient podClient, AuthSession authSession, BdkConfig config) {
+        super(agentClient, podClient, authSession, config);
+        this.started.set(false);
+        this.datafeedId = null;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void start() throws Throwable {
+        this.datafeedId = this.retrieveDatafeedIdFromDisk();
+
+        if (this.datafeedId == null) {
+            this.datafeedId = this.createDatafeedAndSaveToDisk();
+        }
+
+        log.debug("Start reading events from datafeed {}", datafeedId);
+        do {
+            this.started.set(true);
+            this.readDatafeed();
+        } while (this.started.get());
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void stop() {
+        this.started.set(false);
+        this.datafeedId = null;
+    }
+
+    private void readDatafeed() throws Throwable {
+        this.retry.executeCheckedSupplier(() -> {
+            try {
+                List<V4Event> events = datafeedApi.v4DatafeedIdReadGet(datafeedId, authSession.getSessionToken(), authSession.getKeyManagerToken(), null);
+                if (events != null && !events.isEmpty()) {
+                    handleV4EventList(events);
+                }
+            } catch (ApiException e) {
+                log.error("Error {}: {}", e.getCode(), e.getMessage());
+                if (e.isUnauthorized()) {
+                    log.info("Re-authenticate and try again");
+                    authSession.refresh();
+                }
+                if (e.isClientError()) {
+                    log.info("Recreate a new datafeed and try again");
+                    datafeedId = this.createDatafeedAndSaveToDisk();
+                }
+                throw e;
```

Comment:
same as below 
_so if auth has expired it means we still throw the exception and go into the retry from resilience4j and wait before retrying the request where we could retry directly_

Reply:
I'm a bit confused now by the retry strategy. This strategy was suggested by Darlys, and I'm also not sure if we can have two different config in just one Retry instance.